src/containers/withAuthorizedAction/withAuthorizedAction.tsx

```diff
@@ -0,0 +1,163 @@
+import { RootStateOrAny, connect } from 'react-redux'
+import React, { ComponentProps, useCallback, useState } from 'react'
+import { BigNumber, ethers } from 'ethers'
+import {
+  Authorization,
+  AuthorizationType
+} from '../../modules/authorization/types'
+import { getNetworkProvider } from '../../lib/eth'
+import { getAddress } from '../../modules/wallet/selectors'
+import {
+  AuthorizationModal,
+  AuthorizationStepStatus,
+  AuthorizedAction
+} from './AuthorizationModal'
+import {
+  WithAuthorizedActionProps,
+  MapStateProps,
+  AuthorizeActionOptions
+} from './withAuthorizedAction.types'
+import { getERC20ContractInstance, getERC721ContractInstance } from './utils'
+
+const mapState = (state: RootStateOrAny): MapStateProps => ({
+  address: getAddress(state)
+})
+
+export default function withAuthorizedAction<
+  P extends WithAuthorizedActionProps
+>(
+  WrappedComponent: React.ComponentType<P>,
+  action: AuthorizedAction,
+  getConfirmationStatus?: (state: RootStateOrAny) => AuthorizationStepStatus,
+  getConfirmationError?: (state: RootStateOrAny) => string | null
+): React.ComponentType<Omit<P, keyof WithAuthorizedActionProps>> {
+  // TODO: Remove any type
+  const WithAutorizedActionComponent = (props: MapStateProps & any) => {
```

Comment:
could work the `Record<string, unknown>` instead of the any?

Reply:
It doesn't work with unknown  the rendering of the component fails

```
Type '{ onAuthorizedAction: (authorizeOptions: AuthorizeActionOptions) => Promise<void>; isLoadingAuthorization: boolean; address: string | undefined; }' is not assignable to type 'P'.
  '{ onAuthorizedAction: (authorizeOptions: AuthorizeActionOptions) => Promise<void>; isLoadingAuthorization: boolean; address: string | undefined; }' is assignable to the constraint of type 'P', but 'P' could be instantiated with a different subtype of constraint 'WithAuthorizedActionProps'.
```